src/screens/__tests__/LocationDetailsScreen.test.tsx

```diff
@@ -0,0 +1,69 @@
+import {
+  render,
+  screen,
+  fireEvent,
+  waitFor,
+} from "@testing-library/react-native";
+import { Provider } from "@ant-design/react-native";
+
+import { LocationDetailsScreen } from "@screens/LocationDetailsScreen";
+
+import "@testing-library/jest-native/extend-expect";
+
+// Test Utils
+const confirmButtonId = "confirmWaitTimeButton";
+const submitButtonId = "submitWaitTimeButton";
+const getConfirmWaitTimeButton = () => screen.getByTestId(confirmButtonId);
+const getSubmitWaitTimeButton = () => screen.getByTestId(submitButtonId);
+
+describe("<LocationDetailsScreen />", () => {
+  const mockNavigate = jest.fn();
+  const navigation: any = {
+    navigate: mockNavigate,
+    addListener: jest.fn().mockImplementation((event, callback) => {
+      let e = { data: { action: "something" }, preventDefault: jest.fn() };
+      callback(e);
+    }),
+    goBack: jest.fn(),
+  };
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("confirm success modal pops up when user confirms wait time", async () => {
+    render(
+      <Provider>
+        <LocationDetailsScreen navigation={navigation} />
+      </Provider>
+    );
+    fireEvent.press(getConfirmWaitTimeButton());
```

Comment:
Just a note: I haven't figured out mocking the api calls yet but that will have to be added too once you hook up the calls.

Reply:
Yessir